test/cpp/util/channelz_sampler.cc

```diff
@@ -0,0 +1,512 @@
+      for (const auto& _topchannel : get_top_channels_response.channel()) {
+        top_channels.push_back(_topchannel);
+        all_channels.push_back(_topchannel);
+        StoreChannelInJson(_topchannel);
+      }
+      if (!get_top_channels_response.end()) {
+        channel_start_id = GetChannelID(top_channels.back()) + 1;
+      } else {
+        break;
+      }
+    }
+    std::cout << "Number of top channels = " << top_channels.size()
+              << std::endl;
+    return true;
+  }
+
+  // layer traverse for each top channel
+  void TraverseTopChannels() {
+    for (const auto& _topchannel : top_channels) {
+      int tree_depth = 0;
+      std::queue<grpc::channelz::v1::Channel> channel_queue;
+      std::queue<grpc::channelz::v1::Subchannel> subchannel_queue;
+      std::cout << "Tree depth = " << tree_depth << std::endl;
+      GetChannelDescedence(_topchannel, channel_queue, subchannel_queue);
+
+      while (!channel_queue.empty() || !subchannel_queue.empty()) {
+        ++tree_depth;
+        std::cout << "Tree depth = " << tree_depth << std::endl;
+        int ch_q_size = channel_queue.size();
+        int subch_q_size = subchannel_queue.size();
+        for (int i = 0; i < ch_q_size; ++i) {
+          grpc::channelz::v1::Channel ch = channel_queue.front();
+          channel_queue.pop();
+          GetChannelDescedence(ch, channel_queue, subchannel_queue);
+        }
+        for (int i = 0; i < subch_q_size; ++i) {
+          grpc::channelz::v1::Subchannel subch = subchannel_queue.front();
+          subchannel_queue.pop();
+          GetSubchannelDescedence(subch, channel_queue, subchannel_queue);
+        }
+      }
+    }
+  }
+
+  // dump data of all entities to stdout
+  void DumpStdout() {
+    std::string data_str;
+    for (const auto& _channel : all_channels) {
+      std::cout << "channel " << GetChannelID(_channel)
+                << " data:" << std::endl;
+      ::google::protobuf::TextFormat::PrintToString(_channel.data(), &data_str);
+      printf("%s", data_str.c_str());
+    }
+    for (const auto& _subchannel : all_subchannels) {
+      std::cout << "subchannel " << GetSubchannelID(_subchannel)
+                << " data:" << std::endl;
+      ::google::protobuf::TextFormat::PrintToString(_subchannel.data(),
+                                                    &data_str);
+      printf("%s", data_str.c_str());
+    }
+    for (const auto& _server : all_servers) {
+      std::cout << "server " << GetServerID(_server) << " data:" << std::endl;
+      ::google::protobuf::TextFormat::PrintToString(_server.data(), &data_str);
+      printf("%s", data_str.c_str());
+    }
+    for (const auto& _socket : all_sockets) {
+      std::cout << "socket " << GetSocketID(_socket) << " data:" << std::endl;
+      ::google::protobuf::TextFormat::PrintToString(_socket.data(), &data_str);
+      printf("%s", data_str.c_str());
+    }
+  }
+
+  // Clear stored entities to prepare for the next scan
+  void ClearEntities() {
```

Comment:
This can be removed if we manually create a new object each loop iteration

Reply:
Removed